flow/Arena.h

```diff
@@ -313,6 +313,23 @@ class Optional : public ComposedIdentifier<T, 0x10> {
 	bool valid;
 };
 
+// This class is necessary because Optional<T> is effectively trivially
+// destructible if T is trivially destructible, but we can't SFINAE out
+// destructors until C++20, so we still define the destructor for
+// Optional<T> even if it's trivial
+template <class T>
+struct trivially_destructible : public std::is_trivially_destructible<T> {};
+
+template <class T>
+struct trivially_destructible<Optional<T>> : public trivially_destructible<T> {};
+
+// If std::is_trivially_destructible_v<T> is false only because T has a field
+// Optional<U> where U is trivially destructible, we can manually specify that
+// T is trivially_destructible
+#define TRIVIALLY_DESTRUCTIBLE(T)                                                                                      \
```

Comment:
IMO it's just as readable to use `template <> struct trivially_destructible<T> : std::true_type {}` instead of the macro, but either way sounds good to me